editor/sidebar/block-inspector/generic-controls.js

```diff
@@ -16,32 +16,52 @@ import { __ } from '@wordpress/i18n';
 import { updateBlockAttributes } from '../../actions';
 import { getSelectedBlock } from '../../selectors';
 
-class BlockInspectorClassName extends Component {
+/**
+ * Internal constants
+ */
+const ANCHOR_REGEX = /[^\w:.-]/g;
```

Comment:
Seems that older specifications had restrictions, but as of HTML5 is not as strict:

https://www.w3.org/TR/2016/REC-html51-20161101/dom.html#the-id-attribute

Or is the hashbang behavior dependent on this?